### PR TITLE
Code cleanup

### PR DESCRIPTION
**Pull request by AndriySvyryd**

### CODE REVIEW

File: EFCore.sln.DotSettings [hunk 1]
```diff
@@ -1 +0,0 @@
-﻿<wpf:ResourceDictionary xml:space="preserve" xmlns:x="http://schemas.microsoft.com/winfx/2006/xaml" xmlns:s="clr-namespace:System;assembly=mscorlib" xmlns:ss="urn:shemas-jetbrains-com:settings-storage-xaml" xmlns:wpf="http://schemas.microsoft.com/winfx/2006/xaml/presentation" />
```
Comment (smitpatel) [inline EFCore.sln.DotSettings]: Is this file deleted? Is this file needed?

File: src/EFCore.InMemory/Query/Internal/InMemoryExpressionTranslatingExpressionVisitor.cs [hunk 2]
```diff
@@ -124,24 +127,25 @@ protected override Expression VisitConditional(ConditionalExpression conditional
...
             {
                 return result;
             }
 
             static bool shouldApplyNullProtectionForMemberAccess(Type callerType, string memberName)
                 => !(callerType.IsGenericType
-                    && callerType.GetGenericTypeDefinition() == typeof(Nullable<>)
-                    && (memberName == nameof(Nullable<int>.Value) || memberName == nameof(Nullable<int>.HasValue)));
+                     && callerType.GetGenericTypeDefinition() == typeof(Nullable<>)
+                     && (memberName == nameof(Nullable<int>.Value) || memberName == nameof(Nullable<int>.HasValue)));
```
Comment (smitpatel) [inline src/EFCore.InMemory/Query/Internal/InMemoryExpressionTranslatingExpressionVisitor.cs]: why || did not go to new line?

File: src/EFCore.Relational/Diagnostics/IDbTransactionInterceptor.cs [hunk 3]
```diff
@@ -117,7 +119,8 @@ public interface IDbTransactionInterceptor : IInterceptor
         /// <param name="connection"> The connection. </param>
         /// <param name="eventData"> Contextual information about connection and transaction. </param>
         /// <param name="result">
-        ///     The result of the call to <see cref="M:System.Data.Common.DbConnection.BeginTransactionAsync(System.Data.IsolationLevel,System.Threading.CancellationToken)" />.
+        ///     The result of the call to
```
Comment (smitpatel) [inline src/EFCore.Relational/Diagnostics/IDbTransactionInterceptor.cs]: Is this line break intended?

File: src/EFCore.Relational/Query/Internal/SqlExpressionOptimizingExpressionVisitor.cs [hunk 4]
```diff
@@ -60,23 +61,26 @@ protected virtual Expression VisitSqlUnaryExpression(SqlUnaryExpression sqlUnary
                             return SqlExpressionFactory.Constant(constantOperand.Value == null, sqlUnaryExpression.TypeMapping);
 
                         case ColumnExpression columnOperand
-                        when !columnOperand.IsNullable:
+                            when !columnOperand.IsNullable:
```
Comment (smitpatel) [inline src/EFCore.Relational/Query/Internal/SqlExpressionOptimizingExpressionVisitor.cs]: I think VS aligns them rather than tabbed.

File: src/EFCore.Relational/Query/RelationalShapedQueryCompilingExpressionVisitor.RelationalCommandCache.cs [hunk 5]
```diff
@@ -59,8 +60,8 @@ public virtual IRelationalCommand GetRelationalCommand(IReadOnlyDictionary<strin
                     try
                     {
                         var optimizeResult = _parameterValueBasedSelectExpressionOptimizer.Optimize(_selectExpression, parameters);
-                        var selectExpression = 
-                        relationalCommand = _querySqlGeneratorFactory.Create().GetCommand(optimizeResult.selectExpression);
+                        var selectExpression =
```
Comment (smitpatel) [inline src/EFCore.Relational/Query/RelationalShapedQueryCompilingExpressionVisitor.RelationalCommandCache.cs]: Massive typo in actual code.

File: src/EFCore.Relational/Update/ModificationCommand.cs [hunk 6]
```diff
@@ -192,16 +192,16 @@ public virtual void AddEntry([NotNull] IUpdateEntry entry, bool mainEntry)
         private void ValidateState(IUpdateEntry mainEntry, IUpdateEntry entry)
         {
             var mainEntryState = mainEntry.SharedIdentityEntry == null
-                                ? mainEntry.EntityState
-                                : EntityState.Modified;
+                ? mainEntry.EntityState
```
Comment (smitpatel) [inline src/EFCore.Relational/Update/ModificationCommand.cs]: This and SqlExpressionFactory is somewhat conflicting style.
Do we tab ternary from starting point of starting of ternary in =

File: test/EFCore.Cosmos.FunctionalTests/Query/OwnedQueryCosmosTest.cs [hunk 7]
```diff
@@ -169,9 +169,11 @@ public override Task Navigation_rewrite_on_owned_reference_followed_by_regular_e
         }
 
         [ConditionalTheory(Skip = "LeftJoin #12086")]
-        public override Task Navigation_rewrite_on_owned_reference_followed_by_regular_entity_and_another_reference_in_predicate_and_projection(bool isAsync)
+        public override Task
+            Navigation_rewrite_on_owned_reference_followed_by_regular_entity_and_another_reference_in_predicate_and_projection(bool isAsync)
```
Comment (smitpatel) [inline test/EFCore.Cosmos.FunctionalTests/Query/OwnedQueryCosmosTest.cs]: At some point @maumar  put them on 1 line.

File: EFCore.sln.DotSettings (hunk 1 above)
Comment (AndriySvyryd) [reply EFCore.sln.DotSettings]: Deleted, we only need one and it should be named the same as the solution.

File: src/EFCore.Relational/Diagnostics/IDbTransactionInterceptor.cs (hunk 3 above)
Comment (AndriySvyryd) [reply src/EFCore.Relational/Diagnostics/IDbTransactionInterceptor.cs]: Yes, the line is too long

File: src/EFCore.Relational/Query/RelationalShapedQueryCompilingExpressionVisitor.RelationalCommandCache.cs (hunk 5 above)
Comment (AndriySvyryd) [reply src/EFCore.Relational/Query/RelationalShapedQueryCompilingExpressionVisitor.RelationalCommandCache.cs]: Code cleanup __is__ useful 😄

File: src/EFCore.Relational/Update/ModificationCommand.cs (hunk 6 above)
Comment (AndriySvyryd) [reply src/EFCore.Relational/Update/ModificationCommand.cs]: The difference in SqlExpressionFactory is that the expression is in parenthesis, so I don't think we should visually break out.
Generally it's better not to align to arbitrary points as renaming the variables would cause the rest to shift.

File: test/EFCore.Cosmos.FunctionalTests/Query/OwnedQueryCosmosTest.cs (hunk 7 above)
Comment (AndriySvyryd) [reply test/EFCore.Cosmos.FunctionalTests/Query/OwnedQueryCosmosTest.cs]: Yes, it's unfortunate, but we need to try to limit the line length

File: src/EFCore.Relational/Query/Internal/SqlExpressionOptimizingExpressionVisitor.cs (hunk 4 above)
Comment (AndriySvyryd) [reply src/EFCore.Relational/Query/Internal/SqlExpressionOptimizingExpressionVisitor.cs]: I don't see an option in R# to align them or an option in VS to indent them 😕 
@ajcvickers Any preference?

File: src/EFCore.InMemory/Query/Internal/InMemoryLinqOperatorProvider.cs [hunk 8]
```diff
@@ -160,15 +161,20 @@ static InMemoryLinqOperatorProvider()
             LongCountWithoutPredicate = enumerableMethods.Single(
                 mi => mi.Name == nameof(Enumerable.LongCount) && mi.GetParameters().Length == 1);
             LongCountWithPredicate = enumerableMethods.Single(
-                mi => mi.Name == nameof(Enumerable.LongCount) && mi.GetParameters().Length == 2 && IsFunc(mi.GetParameters()[1].ParameterType));
+                mi => mi.Name == nameof(Enumerable.LongCount) && mi.GetParameters().Length == 2
+                                                              && IsFunc(mi.GetParameters()[1].ParameterType));
```
Comment (smitpatel) [inline src/EFCore.InMemory/Query/Internal/InMemoryLinqOperatorProvider.cs]: Why does it not align underneath mi.name like others?

File: src/EFCore.Relational/Migrations/Internal/MigrationsModelDiffer.cs [hunk 9]
```diff
@@ -990,9 +989,7 @@ protected virtual IEnumerable<MigrationOperation> Diff([NotNull] IProperty sourc
 
             var operation = new AddColumnOperation
             {
-                Schema = targetEntityType.GetSchema(),
-                Table = targetEntityType.GetTableName(),
-                Name = target.GetColumnName()
+                Schema = targetEntityType.GetSchema(), Table = targetEntityType.GetTableName(), Name = target.GetColumnName()
```
Comment (smitpatel) [inline src/EFCore.Relational/Migrations/Internal/MigrationsModelDiffer.cs]: The previous is more readable.

File: src/EFCore.Relational/Query/RelationalShapedQueryCompilingExpressionVisitor.RelationalCommandCache.cs [hunk 10]
```diff
@@ -89,8 +85,7 @@ public CommandCacheKey(SelectExpression selectExpression, IReadOnlyDictionary<st
 
                 public override bool Equals(object obj)
                     => obj != null
-                       && (ReferenceEquals(this, obj)
-                           || obj is CommandCacheKey commandCacheKey
+                       && (obj is CommandCacheKey commandCacheKey
```
Comment (smitpatel) [inline src/EFCore.Relational/Query/RelationalShapedQueryCompilingExpressionVisitor.RelationalCommandCache.cs]: Why did we remove reference equality?

File: test/EFCore.Specification.Tests/CustomConvertersTestBase.cs [hunk 11]
```diff
@@ -390,13 +390,8 @@ public virtual void Field_on_derived_type_retrieved_via_cast_applies_value_conve
...
                 var query = context.Set<Blog>()
-                                .Where(b => b.BlogId == 2)
-                                .Select(x => new
-                                {
-                                    BlogId = x.BlogId,
-                                    Url = x.Url,
-                                    RssUrl = x is RssBlog ? ((RssBlog)x).RssUrl : null
-                                }).ToList();
+                    .Where(b => b.BlogId == 2)
+                    .Select(x => new { x.BlogId, x.Url, RssUrl = x is RssBlog ? ((RssBlog)x).RssUrl : null }).ToList();
```
Comment (smitpatel) [inline test/EFCore.Specification.Tests/CustomConvertersTestBase.cs]: The other select.

File: test/EFCore.Specification.Tests/GraphUpdatesFixtureBase.cs [hunk 12]
```diff
@@ -431,8 +431,7 @@ protected virtual object CreateFullGraph()
                                 AlternateId = Guid.NewGuid(),
                                 Children = new ObservableHashSet<RequiredAk2>(ReferenceEqualityComparer.Instance)
                                 {
-                                    new RequiredAk2 { AlternateId = Guid.NewGuid() },
-                                    new RequiredAk2 { AlternateId = Guid.NewGuid() }
+                                    new RequiredAk2 { AlternateId = Guid.NewGuid() }, new RequiredAk2 { AlternateId = Guid.NewGuid() }
```
Comment (smitpatel) [inline test/EFCore.Specification.Tests/GraphUpdatesFixtureBase.cs]: Grass is greener on other side.

File: src/EFCore.InMemory/Query/Internal/InMemoryLinqOperatorProvider.cs (hunk 8 above)
Comment (AndriySvyryd) [reply src/EFCore.InMemory/Query/Internal/InMemoryLinqOperatorProvider.cs]: There are a couple that don't, seems like a bug. Either way I'll send another PR to remove most of this class in favor of EnumerableMethods

File: src/EFCore.Relational/Query/RelationalShapedQueryCompilingExpressionVisitor.RelationalCommandCache.cs (hunk 10 above)
Comment (AndriySvyryd) [reply src/EFCore.Relational/Query/RelationalShapedQueryCompilingExpressionVisitor.RelationalCommandCache.cs]: Value type, no point

File: test/EFCore.Specification.Tests/CustomConvertersTestBase.cs (hunk 11 above)
Comment (AndriySvyryd) [reply test/EFCore.Specification.Tests/CustomConvertersTestBase.cs]: Could you elaborate?

Comment (smitpatel) [reply test/EFCore.Specification.Tests/CustomConvertersTestBase.cs]: A lot of new expressions are getting inline. For short one may be for something like above, RssUrl should be on new line.

Comment (AndriySvyryd) [reply test/EFCore.Specification.Tests/CustomConvertersTestBase.cs]: I changed some rules and reran cleanup

File: test/EFCore.Specification.Tests/GraphUpdatesFixtureBase.cs (hunk 12 above)
Comment (AndriySvyryd) [reply test/EFCore.Specification.Tests/GraphUpdatesFixtureBase.cs]: Splitting all initializations with two members looks terrible.

File: src/EFCore.Design/Design/OperationExecutor.cs [hunk 13]
```diff
@@ -133,7 +133,7 @@ private DatabaseOperations DatabaseOperations
         /// <summary>
         ///     Represents an operation to add a new migration.
         /// </summary>
-        public class AddMigration : OperationBase
+        public class AddMigration : OperationExecutor.OperationBase
```
Comment (smitpatel) [inline src/EFCore.Design/Design/OperationExecutor.cs]: Code change rather than just cleanup?

Comment (AndriySvyryd) [reply src/EFCore.Design/Design/OperationExecutor.cs]: Weird R# bug
